### PR TITLE
Updating yuhow's note

### PR DESCRIPTION
**Pull request by yuhow**
status: still working on noting the detail of how I develop my sod shock tube solver.
Expected time of further updating: 2015/1/21 (Thur.)

### CODE REVIEW

**Comment by tai271828** [comment]
好棒棒

**Comment by yungyuc** [comment]
@tai271828 How about using English for comments?  Although all active contributors now speak Mandarin, it's highly likely to have non-Chinese speaking contributors.

**Comment by yungyuc** [comment]
@yuhow, I think it's OK to merge this PR, unless you want to wait for further editing.  I'll let you to do the merge yourself.

File: yuhow/Makefile [hunk 1]
```diff
@@ -1,76 +0,0 @@
-SHELL := bash
```
Comment (yungyuc) [inline yuhow/Makefile:1]: Why do you remove the make file?  Usually an automatic build system would help people understand how to build.  Also, this make file takes care of toc building and keeps build logs.  Would you please describe why you don't want to use the automatic build script?

File: yuhow/README [hunk 2]
```diff
@@ -0,0 +1,2 @@
+RECOMMENDED COMMAND:
+pdflatex yuhowNote.tex
```
Comment (yungyuc) [inline yuhow/README]: A single run doesn't correctly produce toc files.

File: yuhow/pstake.py [hunk 3]
```diff
@@ -1,806 +0,0 @@
-#!/usr/bin/env python
```
Comment (yungyuc) [inline yuhow/pstake.py:1]: The reason I use PSTricks is it allows us to keep the plain-text source files of the schematics.  It's OK to store binary files like pptx, eps, or pdf in the repository, but they either require a dedicated application or cannot be easily edited.  Could you elaborate why you choose to not use PSTricks?

File: yuhow/yuhowNote.tex [hunk 4]
```diff
@@ -1,4 +1,5 @@
-\documentclass[a4paper,12pt,dvips]{article}
+%\documentclass[a4paper,12pt,dvips]{article}
+\documentclass[a4paper,12pt]{article}
```
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: Why would you choose to use pdflatex instead of latex + dvips?

File: yuhow/yuhowNote.tex [hunk 5]
```diff
@@ -33,27 +82,156 @@
...
-  \item Why the total flux leaving any conservation element is zero? This question is relevant to the definition of Eq. (2.28).
-  \item Why does the term "$-\mu\partial^{2}u^{*}(x, t; j, n)/\partial x^{2} $ vanish "in Eq. (2.29)?
-  \item (In page 303) Why is the condition that Eq. (2.30) being valid uniformly within an SE stronger than Eq. (2.28) for a higher-order expansion? Are they the same thing but just in different form?
-  \item Eq. (2.33) maybe is wrong. The partial derivative in the left-hand side of Eq. (2.33) should be with respect to x instead of t. Will try to double check with author though email. (STATUS: waiting for Prof. Chang's reply)
+  \item Derivation of Eq.(2.5) $(u_t)^n_j = -a(u_x)^n_j$:
+
+        \hspace{4mm}Substituting Eq.(2.3) into Eq.(2.1), then you can get Eq.(2.5).
+
+  \item Derivation of Eq.(2.12) $\frac{4}{(\Delta x)^{2}}F_{\pm}(j, n)=\pm(\frac{1}{2})[(1-\nu^{2}+\xi)(u_{x})^{n}_{j}+(1-\nu^{2}-\xi)(u_{x})^{n-1/2}_{j\pm1/2}]+\frac{2(1\mp\nu)}{\Delta x}(u^{n}_{j}-u^{n-1/2}_{j\pm1/2})$, where $\nu\equiv\frac{a\Delta t}{\Delta x}$ and $\xi\equiv\frac{4\mu\Delta t}{(\Delta x)^{2}}$:
```
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: Long lines are difficult to be reviewed and commented on.  It is usually good to break a line shorter.  A rule of thumb is to [keep each line shorter than 79 characters](https://www.python.org/dev/peps/pep-0008/#maximum-line-length).

Having said that, it's generally accepted that reviewing LaTeX source is difficult if not impractical.  I think the bottom line is that we keep a common source style across all notes in this repository.  @yuhow I guess you've read through the existing notes, like https://github.com/solvcon/cesenote/blob/master/stcon/onedim_old.tex ?  If you want to change any of the style, please point it out and let's change it.

File: yuhow/yuhowNote.tex [hunk 6]
```diff
@@ -62,11 +240,261 @@ \section{Something about Python in Solvcon}
...
   \item iPython
  \end{enumerate}
+And the complete Python code of CESE solver can be found in Section~\ref{subsec:python_code}. Section~\ref{sec:1d_shock_tube} will  demonstrate how we implement CESE solver for solving this one-dimensional shock tube problem in Python.
+
+\subsection{One-dimensional shock tube problem}
+ \label{sec:1d_shock_tube}
+
+The one-dimensional Sod shock tube is employed for the test of our CESE computational fluid codes. Figure~\ref{fig:sod_shock_tube_1D} shows the initial condition of different region in our Sod shock tube. The whole one-dimension space is separated into two regions by a diaphragm in the beginning. 
+The left region is called driven section and containing the gas with the higher density $\rho_l$ and the higher pressure $p_l$. The right region is called working section and with the gas at the lower density $\rho_r$ and the lower pressure $p_r$. Before removing the diaphragm, the gas in both regions is static. 
+The global heat capacity ratio ($\gamma\equiv\frac{C_p}{C_v}$) is 1.4.
```
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: $\gamma = \frac{c_p}{c_v}$ is called specific heat ratio.  "Heat capacity ratio" is a wrong name.  $c_p$ is specific heat at constant pressure while $c_v$ is specific heat at constant volume.  Note it's lower-cased "c", not upper-case "C".  See [1].

I don't think $\equiv$ is a correct operator in this context.  I think $\buildrel{\text{def}}\over{=}$ is more proper for defining a symbol.

[1] J. D. Anderson, Modern Compressible Flow: With Historical Perspective, 3rd ed. Boston: McGraw-Hill, 2003. http://www.amazon.com/Modern-Compressible-Flow-Historical-Perspective/dp/0072424435

File: yuhow/yuhowNote.tex [hunk 7]
```diff
@@ -33,27 +82,156 @@
...
+
+        \hspace{4mm}Eq.(2.6) $u^{*}(x,t;j,n)=u^{n}_{j}+(u_{x})^{n}_{j}[(x-x_{j})-a(t-t^{n})], (x,t)\in SE(j,n)$
+
+        \hspace{4mm}Eq.(2.9) $\vec{\bf{g}}^{*}\equiv(-u^{*},au^{*}-\mu\partial u^{*}/\partial x), d{\bf{r}}\equiv(dx,dt)$
+
+        \hspace{4mm}$F_{\pm}(j,n)=\oint^{c.c.}_{S(CE_{\pm}(j,n))}\vec{\bf{g}}^{*}\cdot d{\bf{r}}$
+
+        \hspace{4mm}For CE$_{-}(ABPC)$ shown in Figure~\ref{fig:mesh_and_ce_se_v1}:
+
+        \hspace{8mm}$\int^{B}_{A}\{-u^{n}_{j}-(u_{x})^{n}_{j}[(x-x_{j})-a(t-t^{n})]\}dx$
```
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: These equations squeeze together in the page.  They aren't easy to read.

File: yuhow/Makefile (hunk 1 above)
Comment (yuhow) [inline yuhow/Makefile:1]: umm...maybe the reason was quite stupid. All the figures I had were pdf format. It was not possible to include pdf figure in Latex. So I just used PdfLatex instead of Latex. One thing after another, I also didn't know how to modify that Makefile. So I just directly abandoned that Makefile.

File: yuhow/README (hunk 2 above)
Comment (yuhow) [inline yuhow/README]: Yes, it took two run to produce correct output file in the beginning.

File: yuhow/yuhowNote.tex (hunk 4 above)
Comment (yuhow) [inline yuhow/yuhowNote.tex]: Umm... it was about the format of input figures. I chose PDF format.

File: yuhow/yuhowNote.tex (hunk 5 above)
Comment (yuhow) [inline yuhow/yuhowNote.tex]: OK. I think your suggestion is very useful. I will follow this basic style rule.

File: yuhow/yuhowNote.tex (hunk 6 above)
Comment (yuhow) [inline yuhow/yuhowNote.tex]: Thanks! I will modify it.

File: yuhow/yuhowNote.tex (hunk 7 above)
Comment (yuhow) [inline yuhow/yuhowNote.tex]: I see. I will tune the distance of each equation.

File: yuhow/Makefile (hunk 1 above)
Comment (yuhow) [inline yuhow/Makefile:1]: NEWS: I will use make file again. I have convert all my pdf figure into eps format.

Comment (yungyuc) [inline yuhow/Makefile:1]: If there's any issue you find in the Makefile we can work it out together.  The file itself isn't particularly complex.  I hope we can make good use of the automation.

File: yuhow/yuhowNote.tex (hunk 7 above)
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: I think display format is more readable than the inline format.  Although we start from scratch, the end results hopefully can be useful for later readers.  The inline math is pretty hard to maintain in LaTeX.

File: yuhow/Makefile (hunk 1 above)
Comment (yuhow) [inline yuhow/Makefile:1]: Actually, it worked fine. I decided to convert all of my figure into EPS format. Everything is OK now.

File: yuhow/pstake.py (hunk 3 above)
Comment (yuhow) [inline yuhow/pstake.py:1]: Yes. I agree with your point of view. There is no any specific reason. In the past, I used ppt to make fancy diagram and converted it into PDF file. Also I used PdfLatex to compile my tex file.
Anyway, I will start to use PSTricks for making diagram. It's now that hard to do that. Once I have PDF file, some software can help me to convert it into EPS format.

File: yuhow/Makefile (hunk 1 above)
Comment (yungyuc) [inline yuhow/Makefile:1]: EPS will resolve the pdflatex issue.  Moving back to latex+dvips allows us to use both EPS and PSTricks with PStake.  It is good.

A further step is how would we collaborating on drawings.  For mesh_and_ce_se_v1.pptx, I have https://github.com/solvcon/cesenote/blob/master/stcon/schematic/staggered_stencil.tex that you might find similar.  For sod_shock_tube_1D.pptx, I have something in my laptop that should have been shared.  They are both in PSTricks/LaTeX format, and can be converted to EPS by PStake.

File: yuhow/yuhowNote.tex (hunk 4 above)
Comment (yungyuc) [inline yuhow/yuhowNote.tex]: I suppose you [are using dvips now](https://github.com/yuhow/cesenote/commit/6470db233dcc2b5a94043feb7771de2e03497d2e#diff-3505381d264d0e9bd5b841f26866559cR50).  In this case it would be clear to revert this change to explicitly specify the driver as it did.

**Comment by yuhow** [comment]
Just updating my new solver and also its unit test.

**Comment by yungyuc** [comment]
Can't wait :)
